phpBB/phpbb/search/backend/base.php

```diff
@@ -382,7 +396,9 @@ public function delete_index(int &$post_counter = null): ?array
 
 		$starttime = microtime(true);
 		$row_count = 0;
-		while (still_on_time() && $post_counter <= $max_post_id)
+
+		$still_on_time = PHP_SAPI === 'cli' ? true : still_on_time();
+		while ($still_on_time && $post_counter < $max_post_id)
```

Comment:
```suggestion
		while ($still_on_time && $post_counter <= $max_post_id)
```
Just a note, that there is one more `while(...)`

Reply:
What I have done is remove it to make it work... I will try to see tomorrow in depth...

Reply:
I will give this a try on Sunday on web interface.

Reply:
Final changes:
```suggestion
		while ((PHP_SAPI === 'cli' ? true : still_on_time()) && $post_counter < $max_post_id)
```

Reply:
did you check this 2 comments : 
1. https://github.com/phpbb/phpbb/pull/5439#discussion_r801023611
2. https://github.com/phpbb/phpbb/pull/5439#discussion_r801628635

Reply:
Oh, you are right, thank you very much
What a stupid mistake 🤦

Reply:
@Dark1z Are you sure is `<=`?, because if `$post_counter = $max_post_id`, `get_posts_batch_after` will return no rows

Reply:
> @Dark1z Are you sure is `<=`?, because if `$post_counter = $max_post_id`, `get_posts_batch_after` will return no rows

Please make it `<` .